jax/_src/pjit.py

```diff
@@ -1572,17 +1572,18 @@ def _resolve_in_shardings(args, pjit_in_shardings: Sequence[PjitSharding]
         # jax.jit does not allow resharding across different memory kinds even
         # if the argument is uncommitted. Use jax.device_put for those cases,
         # either outside or inside jax.jit.
+        assert not isinstance(arg_s, UnspecifiedValue)  # necessary for mypy attribute access below
```

Comment:
Maybe just change on line 1571? `not isinstance(arg_s, UnspecifiedValue)`

Reply:
Should we just inline all instances of `is_unspecified`?

Reply:
Yeah sounds good. We can remove a bunch of type ignores.

But do that in a separate CL? Also you will need to inline `is_auto` and `is_unspecified_or_auto`.

Reply:
Done